pkg/cmd/workflow/view/http.go

```diff
@@ -0,0 +1,76 @@
+package view
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/workflow/shared"
+)
+
+func getWorkflowByID(client *api.Client, repo ghrepo.Interface, workflowID string) (*shared.Workflow, error) {
+	var workflow shared.Workflow
+
+	err := client.REST(repo.RepoHost(), "GET",
+		fmt.Sprintf("repos/%s/actions/workflows/%s", ghrepo.FullName(repo), workflowID),
+		nil, &workflow)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &workflow, nil
+}
+
+func getWorkflowsByName(client *api.Client, repo ghrepo.Interface, workflowName string) ([]shared.Workflow, error) {
+	workflows, err := shared.GetWorkflows(client, repo, 100)
```

Comment:
Are we worried about the 100 limit here? Should we be retrieving all workflows?

Reply:
yeah, we should probably be doing that.